Add a metadata-based certificate search engine

MPSxDev opened the pull request:
### Searchable registry of issuer's certificates  | [Issue 110](https://github.com/zk-passport/proof-of-passport/issues/110)

_Description_

A search engine is added that allows for quick finding of certificate metadata, also indicating if they are supported using the _supported algorithms verifier._

_Changes_

**1. New route /search**

   - File: /search/page.tsx
   - Description: Enables the implementation of the Input and Search components in the web layout.

**2. Create the Search Input component.**

   - File: /components/search/SearchInput.tsx
   - Description: Allows entering a search query that will generate a _searchQuery_ recognized by the _SearchResults_.

**3. Create the Search Results component.**

   - File: /components/search/SearchResults.tsx
   - Description: Delivers search results based on the input in _SearchInput_.

_Features_

The search engine allows searching by entering one or more strings that match any of the included data.

The search result updates immediately when a character is typed into the Input.
![image](https://github.com/user-attachments/assets/5b7ebf00-f34f-4e62-a613-84c08f7d5b97)

1. Simple search: If I enter _"es"_, all certificates containing "es" will appear.
![image](https://github.com/user-attachments/assets/f45f0fb6-6d84-4285-92ad-616ad5af674e)

2. Complex search: If I enter _"es sha1"_, only certificates containing "es" and "sha1" will appear.
![image](https://github.com/user-attachments/assets/7281b3f0-45dc-4b4d-ab4a-b10a10546f37)

3. Complex search: If I enter _"es sha1 2048"_, only certificates containing "es", "sha1", and "2048" will appear.
![image](https://github.com/user-attachments/assets/022f63b7-1fb8-4866-a6cf-5bdd6e85cc2f)

* There is no need for a specific order; I can search for _"2048 es sha1"_ and get the same result as in 3.

MPSxDev committed:
Search engine for certificates

2 components are added for the search engine SearchInput and SearchResults

MPSxDev committed:
Update package.json

0xturboblitz commented:
Hey, apologies for the delay.

Few suggestions:
- When the search box is empty, the whole looks shrunken. It would be better if there was an empty space instead. 
<img width="500" alt="Screenshot 2024-08-15 at 17 16 22" src="https://github.com/user-attachments/assets/f4e572da-3687-4ccd-90ad-324297ee7bf3">
- When I type the name of a country, like "France", it doesn't show the certificates. You can rely on [this list](https://github.com/zk-passport/openpassport/blob/ee00a0244f06b2e1d22b3c3afa29442d66cecc08/common/src/constants/constants.ts#L42) or a similar one for 2-letter country codes for that
- Just like we did with the map, I think emojis like ✅ and 🚧 for the current support make things very clear and distinctive.
- Seems like the thing doesn't make the distinction between cscas and dscs, doesn't show the amount of certificates for each, so ends up not being very useful in its current state

If you have telegram, you can message me at FlorentTavernier for easier chatting

MPSxDev committed:
Api to request certificates for search

api that responds search result in format determined by prisma/Schema searching the database

MPSxDev committed:
Search Engine update

Search engine that queries the /api/certificates?search= returning in json format the certificate(s) that meet the search.
The API queries the Postgres database.

MPSxDev committed:
Search engine update for toggle DSC and CSCA and other features

-Add table to prisma schema.
-Add toggle button for DSC and CSCA.
-Add counter for the result of the query of found certificates.
-Add loading animation while API gets the result.
-Auto-complete for search terms based on 'rsa', 'ecdsa', 'sha', 'ecd', 'pss' to avoid confusion and search quickly with one click.
- Fix inline Modulus.
- Add type parameter for api/certificates type= dsc or csca

MPSxDev committed:
Update yarn.lock

MPSxDev committed:
Merge branch 'dev' into main

remicolin committed:
update padding/margin of toggle button

remicolin committed:
add @prisma/client

remicolin committed:
add keyword based search

remicolin committed:
set max request to 100 certs and fix autocomplete

remicolin committed:
update Autocomplete

remicolin committed:
Merge branch 'dev' into main
